assemblyline/odm/models/ontology/types/antivirus.py

```diff
@@ -0,0 +1,25 @@
+from assemblyline import odm
+
+
+class Antivirus(odm.Model):
+    class Detection(odm.Model):
+        class Engine(odm.Model):
+            class Definition(odm.Model):
+                update_time = odm.Optional(odm.Date())                          # Time of last update
+                version = odm.Optional(odm.Keyword())                           # Version of definition set
+            name = odm.Keyword()                                                # Name of AV engine
+            version = odm.Optional(odm.Keyword())                               # Version of AV engine
+            definition = odm.Optional(odm.Compound(Definition))                 # Details about definition
+
+        # What category does the verdict fall under?
+        category = odm.Optional(odm.Enum(['type-unsupported',                   # File sent to AV is unsupported
+                                          'undetected',                         # File not detected by AV
+                                          'failure',                            # AV failed during detection
+                                          'harmless',                           # AV deems harmless
+                                          'suspicious',                         # AV deems suspicious
+                                          'malicious']))                        # AV deems malicious
+        engine = odm.Compound(Engine)
+        virus_name = odm.Optional(odm.Keyword())                                # The name of the virus
```

Comment:
@cccs-kevin  should this be a list?

Reply:
I don't think so, although it would depend on what our definition of a "Detection" is. If a detection is when a single signature is raised on a file, then it shouldn't be a list. If a detection is all information returned from a scan, then it should be a list...